imagebuildah/chroot_symlink.go

```diff
@@ -24,37 +21,6 @@ func init() {
 	reexec.Register(symlinkModifiedTime, resolveSymlinkTimeModified)
 }
 
-// main() for resolveSymlink()'s subprocess.
-func resolveChrootedSymlinks() {
-	status := 0
-	flag.Parse()
-	if len(flag.Args()) < 2 {
-		fmt.Fprintf(os.Stderr, "%s needs two arguments\n", symlinkChrootedCommand)
-		os.Exit(1)
-	}
-	// Our first parameter is the directory to chroot into.
-	if err := unix.Chdir(flag.Arg(0)); err != nil {
-		fmt.Fprintf(os.Stderr, "chdir(): %v\n", err)
-		os.Exit(1)
-	}
-	if err := unix.Chroot(flag.Arg(0)); err != nil {
-		fmt.Fprintf(os.Stderr, "chroot(): %v\n", err)
-		os.Exit(1)
-	}
-
-	// Our second parameter is the path name to evaluate for symbolic links
-	symLink, err := getSymbolicLink(flag.Arg(1))
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "error getting symbolic links: %v\n", err)
-		os.Exit(1)
-	}
-	if _, err := os.Stdout.WriteString(symLink); err != nil {
-		fmt.Fprintf(os.Stderr, "error writing string to stdout: %v\n", err)
-		os.Exit(1)
-	}
-	os.Exit(status)
-}
-
 // resolveSymlink uses a child subprocess to resolve any symlinks in filename
 // in the context of rootdir.
 func resolveSymlink(rootdir, filename string) (string, error) {
```

Comment:
Move this to `chroot_symlink_linux.go`, along with `resolveChrootedSymlinks()`, and stub just `resolveSymlink()` out in `chroot_symlink_unsupported.go` - `resolveSymlink()` is the function that gets called from elsewhere, and it'd be better to have it return an error on other platforms than to have it try to run a child process that will just mysteriously exit immediately because we're stubbing out its `main()`.